hwy/ops/x86_256-inl.h

```diff
@@ -6300,24 +6300,27 @@ HWY_DIAGNOSTICS(pop)
 
 #endif  // HWY_DISABLE_F16C
 
+#if HWY_AVX3_HAVE_F32_TO_BF16C
 template <class D, HWY_IF_V_SIZE_D(D, 16), HWY_IF_BF16_D(D)>
-HWY_API VFromD<D> DemoteTo(D dbf16, Vec256<float> v) {
-  // TODO(janwas): _mm256_cvtneps_pbh once we have avx512bf16.
-  const Rebind<int32_t, decltype(dbf16)> di32;
-  const Rebind<uint32_t, decltype(dbf16)> du32;  // for logical shift right
-  const Rebind<uint16_t, decltype(dbf16)> du16;
-  const auto bits_in_32 = BitCast(di32, ShiftRight<16>(BitCast(du32, v)));
-  return BitCast(dbf16, DemoteTo(du16, bits_in_32));
+HWY_API VFromD<D> DemoteTo(D /*dbf16*/, Vec256<float> v) {
+  // reinterpret_cast is needed to bit cast the __m128bh vector returned by
+  // _mm256_cvtneps_pbh to a __m128i vector as there is currently no intrinsic
+  // available (as of GCC 13 and Clang 17) to bit cast a __m128bh vector to a
+  // __m128i vector
+  return VFromD<D>{reinterpret_cast<__m128i>(_mm256_cvtneps_pbh(v.raw))};
```

Comment:
Would you like to add a function that does these casts, so we can avoid repeating the comment?

Reply:
I have added detail::BitCastToInteger overloads for __m128bh to __m128i, __m256bh to __m256i, and __m512bh to __m512i on AVX3 targets that support the _mm*_cvtneps_pbh and _mm*_cvtne2ps_pbh intrinsics to avoid repeating the comments.

I have also updated the F32 to BF16 DemoteTo and ReorderDemote2To operations to bit cast the __m128bh to __m128i, __m256bh to __m256i, and __m512bh to __m512i using detail::BitCastToInteger.